asciimatics/screen.py

```diff
@@ -1347,6 +1348,49 @@ def wrapper(cls, func, height=None, catch_interrupt=False, arguments=None,
         finally:
             screen.close(restore)
 
+    class Session():
+        """
+        Decorator and class to create a managed Screen. It can be used in
+        two ways. If used as a method decorator it will create and open a new Screen,
+        pass the screen to the method as a keyword argument, and close the
+        screen when the method has completed. If used with the with statement
+        the class will create and open a new Screen, return the screen for
+        using in the block, and close the screen when the statement ends.
+        Note that any arguments are in this class so that you can use it
+        as a decorator or using the with statment. No arguments are required
+        to use.
+        """
+
+        def __init__(self, func=lambda: None):
+            update_wrapper(self, func)
+            self.func = func
+
+        def __get__(self, obj, objtype):
+            """
+            Get the object (for wrapping class methods)
```

Comment:
The reason you need to do this is not obvious.  Could you please provide a reference to the SO answer that explained how to use this method?

Reply:
I think it was [this](https://stackoverflow.com/a/45361673/3903479) one, I did it so I could use the same object with the `with` statement.